assets/js/theme/page-manager.js

```diff
@@ -1,35 +1,22 @@
-import async from 'async';
+import $ from 'jquery';
 
 export default class PageManager {
     constructor(context) {
         this.context = context;
     }
 
-    before(next) {
-        next();
-    }
-
-    loaded(next) {
-        next();
+    type() {
+        return this.constructor.name;
     }
 
-    after(next) {
-        next();
+    onReady() {
     }
 
-    type() {
-        return this.constructor.name;
-    }
+    static load(context) {
+        const page = new this(context);
```

Comment:
const page = new PageManager(context); ?

new this(context) ? what would that even refer to?

Reply:
This will call the subclass constructor.